### PR TITLE
Update another 16 test modules according to #2776

### PR DESCRIPTION
**Pull request by abalakh**
Part of #2776
Updated following CLI tests:
* location
* medium
* model
* org
* os
* product
* proxy
* report
* repository
* roles
* scparam
* subnet
* subscription
* syncplan
* template
* user

### CODE REVIEW

**Comment by abalakh** [comment]
```
nosetests tests/foreman/cli/test_location.py                    
...S......................S........................................
----------------------------------------------------------------------
Ran 67 tests in 731.804s

OK (SKIP=2)

nosetests tests/foreman/cli/test_medium.py  
.................
----------------------------------------------------------------------
Ran 17 tests in 172.880s

OK

nosetests tests/foreman/cli/test_model.py 
...............................
----------------------------------------------------------------------
Ran 31 tests in 329.140s

OK

nosetests tests/foreman/cli/test_org.py
..SSSSS.S.S.S...S.SSSS.SSSS.SSSS..SSSSS......SSS.SSS................SS..................................SSSSSS......SSSSSS....................................SSSSSS......SSSS.SSSS.SSSS.SSSSSSSSS.SSSS.SSSS
----------------------------------------------------------------------
Ran 204 tests in 1676.229s

OK (SKIP=81)

nosetests tests/foreman/cli/test_os.py 
...........................................S
----------------------------------------------------------------------
Ran 44 tests in 429.731s

OK (SKIP=1)

nosetests tests/foreman/cli/test_product.py
............................................................
----------------------------------------------------------------------
Ran 60 tests in 860.705s

OK

nosetests tests/foreman/cli/test_proxy.py  
SSSSSSSSSSSSSSSSSS
----------------------------------------------------------------------
Ran 18 tests in 0.002s

OK (SKIP=18)

nosetests tests/foreman/cli/test_report.py      
...
----------------------------------------------------------------------
Ran 3 tests in 104.625s

OK

nosetests tests/foreman/cli/test_repository.py
...................................SSSSSS.....................SSSS.....SSSS.
----------------------------------------------------------------------
Ran 76 tests in 622.845s

OK (SKIP=14)

nosetests tests/foreman/cli/test_roles.py         
SSSSSSSSSSSS
----------------------------------------------------------------------
Ran 12 tests in 2.098s

OK (SKIP=12)

nosetests tests/foreman/cli/test_scparam.py
.
----------------------------------------------------------------------
Ran 1 test in 38.609s

OK

nosetests tests/foreman/cli/test_subnet.py
..SSS............................................
----------------------------------------------------------------------
Ran 49 tests in 399.604s

OK (SKIP=3)

nosetests tests/foreman/cli/test_subscription.py
.S....
----------------------------------------------------------------------
Ran 6 tests in 451.964s

OK (SKIP=1)

nosetests tests/foreman/cli/test_syncplan.py    
S........................................................................
----------------------------------------------------------------------
Ran 73 tests in 593.977s

OK (SKIP=1)

nosetests tests/foreman/cli/test_template.py
.........
----------------------------------------------------------------------
Ran 9 tests in 126.615s

OK

nosetests tests/foreman/cli/test_user.py                              
...SSS..SSSSSSS............................................S..............SSSSSSSSSS......SSSSSSSSS...........................S............SSSSSSSSSS.....SSSSSSSS.................SSSSSSSSSSSSSSSSSSSSSS
----------------------------------------------------------------------
Ran 201 tests in 1355.315s

OK (SKIP=71)
```

File: tests/foreman/cli/test_model.py [hunk 1]
```diff
@@ -49,25 +45,10 @@ def test_update_model(self):
...
-        result = Model.info({'name': new_name})
-        self.assertEqual(result.return_code, 0)
-        self.assertEqual(len(result.stderr), 0)
-        self.assertEqual(
-            result.stdout['name'],
-            new_name,
-            "Model name was not updated"
-        )
+        new_name = gen_string('alpha')
+        Model().update({'name': model['name'], 'new-name': new_name})
```
Comment (elyezer) [inline tests/foreman/cli/test_model.py]: Please s/Model()/Model.

File: tests/foreman/cli/test_medium.py [hunk 2]
```diff
@@ -191,22 +132,10 @@ def test_medium_update(self):
...
         """
         new_name = gen_alphanumeric(6)
-        try:
-            medium = make_medium({'name': gen_alphanumeric(6)})
-        except CLIFactoryError as e:
-            self.fail(e)
-
-        args = {
+        medium = make_medium()
+        Medium().update({
```
Comment (elyezer) [inline tests/foreman/cli/test_medium.py]: Please s/Medium()/Medium the entire module.

File: tests/foreman/cli/test_org.py [hunk 3]
```diff
@@ -771,39 +711,30 @@ def test_remove_configtemplate(self, data):
...
+        template = make_template({
+            'content': gen_string('alpha'),
+            'name': name,
+        })
         # Add config-template
-        result = Org.add_config_template({
+        Org.add_config_template({
             'name': org['name'],
-            'config-template': tmplt['name']
+            'config-template': template['name']
```
Comment (elyezer) [inline tests/foreman/cli/test_org.py]: :+1:

File: tests/foreman/cli/test_scparam.py [hunk 4]
```diff
@@ -29,8 +29,4 @@ def test_bugzilla_1047794(self):
         """
 
         self.run_puppet_module()
-        result = SmartClassParameter().list()
-        self.assertEqual(
-            result.return_code, 0, "Command should have succeeded")
-        self.assertEqual(
-            len(result.stderr), 0, "Should not have raised an error")
+        SmartClassParameter().list()
```
Comment (elyezer) [inline tests/foreman/cli/test_scparam.py]: Please s/SmartClassParameter()/SmartClassParameter the entire module.

File: tests/foreman/cli/test_user.py [hunk 5]
```diff
@@ -44,12 +44,8 @@ def __assert_exists(self, user):
         result = UserObj().list({
```
Comment (elyezer) [inline tests/foreman/cli/test_user.py]: Please s/UserObj()/UserObj the entire module.

**Comment by elyezer** [comment]
ACK, the comments are more to update really old code to the new format. It is not a blocker but I think we can use this opportunity to update them.

**Comment by abalakh** [comment]
Agree. Updated

**Comment by elyezer** [comment]
ACK

**Comment by sthirugn** [comment]
ACK

**Pull request merged**